python/ray/tune/tests/test_result_grid.py

```diff
@@ -224,6 +224,26 @@ def f(config):
     assert not any(key in representation for key in AUTO_RESULT_KEYS)
 
 
+def test_result_grid_repr(ray_start_2_cpus):
+    def f(config):
+        with tune.checkpoint_dir(step=0) as checkpoint_dir:
+            path = os.path.join(checkpoint_dir, "checkpoint")
+            with open(path, "w") as f:
+                f.write(json.dumps({"step": 0}))
+
+        from ray.air import session
+
+        session.report({"loss": 1})
+
+    tuner = tune.Tuner(f, param_space={"x": tune.grid_search([1, 2])})
+    result_grid = tuner.fit()
+
+    from ray.tune.result import AUTO_RESULT_KEYS
+
+    representation = result_grid.__repr__()
+    assert not any(key in representation for key in AUTO_RESULT_KEYS)
```

Comment:
Can we check that 2 result reprs get shown in the result grid repr? Checking that all trials get printed?